### PR TITLE
WIP: Add Valhalla PR Build

### PR DESCRIPTION
**Pull request by AdamBrousseau**
Signed-off-by: Talia McCormick <Talia.McCormick@ibm.com>

Correct string comparison

Signed-off-by: Talia McCormick <Talia.McCormick@ibm.com>
update machine label names (external vs internal)

Signed-off-by: Talia McCormick <Talia.McCormick@ibm.com>
Update openjdk repo to openj9 instead of hg and branch

Signed-off-by: Talia McCormick <Talia.McCormick@ibm.com>

point to internal freemarker jar instead of normal freemarker jar

Signed-off-by: Talia McCormick <Talia.McCormick@ibm.com>
update jdk path

Signed-off-by: Talia McCormick <Talia.McCormick@ibm.com>

### CODE REVIEW

**Comment by AdamBrousseau** [comment]
Jenkins test experimental plinux jdkvalhalla

File: buildenv/jenkins/common/pipeline-functions [hunk 1]
```diff
@@ -152,6 +152,9 @@ def workflow() {
                 case "_extended":
                     TEST_JOB_NAME = "Test-Extended-JDK${SDK_VERSION}-${SPEC}"
                     break
+                case "_experimental":
+                    TEST_JOB_NAME = "Test-Experimental-JDK${SDK_VERSION}-${SPEC}"
+                    break
```
Comment (AdamBrousseau) [inline buildenv/jenkins/common/pipeline-functions]: Don't need this for the PR build, only if you plan on adding a standalone test job.

File: buildenv/jenkins/variables/defaults.yml [hunk 2]
```diff
@@ -50,20 +57,58 @@ linux_ppc-64_cmprssptrs_le:
...
+      9: 'ci.project.openj9 hw.arch.ppcle sw.os.rhel7'
+      10: 'ci.project.openj9 hw.arch.ppcle sw.os.rhel7'
+      panama: 'ci.project.openj9 hw.arch.ppcle sw.os.rhel7'
+      valhalla_nestmates: 'ci.project.openj9 hw.arch.ppcle sw.os.rhel7'
+    test:
+      8: 'ci.project.openj9 hw.arch.ppcle sw.os.ubuntu16'
+      9: 'ci.project.openj9 hw.arch.ppcle sw.os.linux'
+      10: 'ci.project.openj9 hw.arch.ppcle sw.os.linux'
+      panama: 'ci.project.openj9 hw.arch.ppcle sw.os.linux'
+      valhalla_nestmates: 'ci.project.openj9 hw.arch.ppcle sw.os.linux'
```
Comment (AdamBrousseau) [inline buildenv/jenkins/variables/defaults.yml]: Will have to update these with the external values

**Comment by AdamBrousseau** [comment]
cc @taliamccormick

File: buildenv/jenkins/common/pipeline-functions [hunk 3]
```diff
@@ -152,6 +152,9 @@ def workflow() {
                 case "_extended":
                     TEST_JOB_NAME = "Test-Extended-JDK${SDK_VERSION}-${SPEC}"
                     break
+                case "_experimental":
```
Comment (smlambert) [inline buildenv/jenkins/common/pipeline-functions]: I do not agree with adding a new top-level test make target.  This implies that all Valhalla test playlist.xml files have been modified also, and would also require some overhaul of test documentation at both OpenJ9 and Adopt.  It also opens the door to others tagging/categorizing tests as experimental, and I am reserving the right to control whether or not we introduce more top-level test make targets.

**Comment by smlambert** [comment]
FYI @llxia

**Comment by taliamccormick** [comment]
@smlambert 

> I do not agree with adding a new top-level test make target, 'make _experimental'. This implies that all Valhalla test playlist.xml files have been modified also, and would also require some overhaul of test documentation at both OpenJ9 and Adopt. It also opens the door to others tagging/categorizing tests as experimental, and I am reserving the right to control whether or not we introduce more top-level test make targets.

Unintentional, sorry - I had missed the connection between `TEST_TARGETS` and the top-level test make target and wanted the job name to indicate a difference between the normal Sanity/Extended tests and the Valhalla/other future tests which are not yet run. I won't add another top level test make target.

From current understanding, the Valhalla tests are compiled under the Sanity make target, using `JAVA_VERSION` Valhalla and `JCL_VERSION` SE90 and I would therefore not need another top level test make target

--

Current goal:
- Add a PR build for valhalla nestmates
- Facilitate addition of PR builds for other valhalla features (eg. valhalla lworld, valhalla ConDy) and for panama

Current plan:
Add a PR build for valhalla nestmates which
a. Compiles a valhalla nestmates jdk using openjdk 10 and openj9 with the valhalla nestmates build flag enabled
b. Run the valhalla tests against the jdk

**Comment by smlambert** [comment]
Thanks for the explanation @taliamccormick

**Comment by llxia** [comment]
yes, there is no need to create PR experimental. In test, valhalla and panama are treated as java version (same as SE80, SE90, etc). I think we can just set JAVA_VERSION=Valhalla, and reuse the existing code. We may not need to use JCL_VERSION.

**Comment by smlambert** [comment]
I would assume you do not need to set JCL_VERSION, as that was introduced as a temporary environment variable to cover off a 'b148' story we were tied to (2 values of JCL_VERSION can be current or latest... for Valhalla, you would likely want the default a.k.a. latest).

**Comment by taliamccormick** [comment]
Jenkins test experimental plinux jdkvalhalla

**Comment by taliamccormick** [comment]
I removed the 'experimental' keyword but am temporarily leaving the job name as-is because I'm not sure if it's tied to the setup that allows me to test here

**Comment by taliamccormick** [comment]
Jenkins test experimental plinux jdkvalhalla

**Comment by taliamccormick** [comment]
Jenkins test experimental plinux jdkvalhalla

**Comment by taliamccormick** [comment]
Jenkins test experimental plinux jdkvalhalla

**Comment by taliamccormick** [comment]
Jenkins test experimental plinux jdkvalhalla

**Comment by taliamccormick** [comment]
Jenkins test experimental plinux jdkvalhalla

**Comment by taliamccormick** [comment]
Jenkins test experimental plinux jdkvalhalla

**Comment by taliamccormick** [comment]
Jenkins test experimental plinux jdkvalhalla

**Comment by taliamccormick** [comment]
Jenkins test experimental plinux jdkvalhalla

**Comment by taliamccormick** [comment]
Jenkins test experimental plinux jdkvalhalla

**Comment by taliamccormick** [comment]
Jenkins test experimental plinux jdkvalhalla

**Comment by taliamccormick** [comment]
Jenkins test experimental plinux jdkvalhalla

**Comment by AdamBrousseau** [comment]
Jenkins test experimental plinux jdkvalhalla

**Comment by AdamBrousseau** [comment]
Jenkins test experimental plinux jdkvalhalla

**Comment by AdamBrousseau** [comment]
Jenkins test experimental plinux jdkvalhalla

**Comment by taliamccormick** [comment]
Jenkins test experimental plinux jdkvalhalla

**Comment by taliamccormick** [comment]
Jenkins test experimental plinux jdkvalhalla

**Comment by taliamccormick** [comment]
Jenkins test experimental plinux jdkvalhalla

**Comment by taliamccormick** [comment]
Jenkins test experimental plinux jdkvalhalla

**Comment by taliamccormick** [comment]
Jenkins test experimental plinux jdkvalhalla

**Comment by taliamccormick** [comment]
Jenkins test experimental plinux jdkvalhalla

**Comment by taliamccormick** [comment]
Jenkins test experimental plinux jdk11-nestmates

**Comment by taliamccormick** [comment]
Jenkins test experimental plinux jdk11-nestmates

**Comment by taliamccormick** [comment]
Jenkins test sanity plinux jdk11-nestmates

**Comment by AdamBrousseau** [comment]
@taliamccormick can I close this or do you still need it?

**Pull request closed without merging**